File: Core/Entities/OrderAggregate/OrderItem.cs
namespace Core.Entities.OrderAggregate
{
    public class OrderItem :BaseEntity 
    {
        public required ProductItemOrdered ItemOrdered { get; set; } // Product details
        public decimal Price { get; set; } // Price of the product at the time of order
        public int Quantity { get; set; } // Quantity of the product ordered
    }
}

File: java-api/core/src/main/java/com/aayuskinet/core/entities/orderaggregate/OrderItem.java
package com.aayuskinet.core.entities.orderaggregate;

import com.aayuskinet.core.entities.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Embedded;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;

// @Entity
@Data
@EqualsAndHashCode(callSuper = true)
public class OrderItem extends BaseEntity {

    @Embedded
    private ProductItemOrdered itemOrdered;

    @Column(precision = 18, scale = 2)
    private BigDecimal price;
    private int quantity;

    @ManyToOne
    @JoinColumn(name = "order_id")
    private Order order;
}
